pkg/kv/bulk/sst_batcher.go

```diff
@@ -123,6 +127,9 @@ type SSTBatcher struct {
 	ms enginepb.MVCCStats
 	// rows written in the current batch.
 	rowCounter storage.RowCounter
+
+	mu  syncutil.Mutex
```

Comment:
can we wrap all the fields we want to lock/unlock in an:
```
mu struct {
syncutil.Mutex
}
```

Reply:
Took a crack at this. It is a little tricky since we read from some of these fields in BufferingAdder but that is after a call to Flush() which called Wait(), so it is mostly just the batcher itself that needs to lock.